### DISCUSSION TITLE
Clearly I'm doing something wrong - I'm getting a glob error that by no means makes sense to me.

### DISCUSSION THREAD

**Discussion by HeatherComputer** [Kira-NT/mc-publish#115] [Q&A] [answered]
I'm trying to migrate to mc-publish after issues with my previous workflows.

I know the exact filename of the jar being uploaded, so I'm giving that directly as a glob in the files parameter:
```yml
files: |
    AdvancedBackups-forge-1.7.10-${{ steps.version.outputs.VERSION }}.jar
 ```
 
In my current case, this fully resolves to `AdvancedBackups-forge-1.7.10-3.5.jar` which *does* have an exact match and is a valid glob string.
I verified this using the DigitalOcean tool you reccomended.

However, mc-publish is spitting out an error:
```
 Error: ArgumentError: No files found for the specified glob. Please ensure the glob is correct and files matching the pattern exist in the specified directory. (Parameter 'options.files')
 ```
 
 The previous step is a simple ls, and it confirms the presence of the target file:
 ```
AdvancedBackups-ALL_JARS-3.5.zip
AdvancedBackups-fabric-1.18-3.5.jar
AdvancedBackups-fabric-1.19.2-3.5.jar
AdvancedBackups-fabric-1.19.3-3.5.jar
AdvancedBackups-fabric-1.20-3.5.jar
AdvancedBackups-fabric-1.20.2-3.5.jar
AdvancedBackups-fabric-1.20.4-3.5.jar
AdvancedBackups-forge-1.12-3.5.jar
AdvancedBackups-forge-1.16-3.5.jar
AdvancedBackups-forge-1.18-3.5.jar
AdvancedBackups-forge-1.19.2-3.5.jar
AdvancedBackups-forge-1.19.3-3.5.jar
AdvancedBackups-forge-1.20-3.5.jar
AdvancedBackups-forge-1.20.2-3.5.jar
AdvancedBackups-forge-1.20.4-3.5.jar
AdvancedBackups-forge-1.7.10-3.5.jar
AdvancedBackups-neoforge-1.20.2-3.5.jar
AdvancedBackups-neoforge-1.20.4-3.5.jar
```
Third from the bottom, that's an exact match to the glob I've provided.

My current mc-publish workflow is visible [here](https://github.com/MommyHeather/AdvancedBackups/blob/core/.github/workflows/publish.yml).

An erroring debug workflow run is visible [here](https://github.com/MommyHeather/AdvancedBackups/actions/runs/8662978256/job/23756097831).

Is there something obvious that I've missed?

**Comment by Kira-NT** [comment, marked as answer]
I was *this* close to having a stroke while I tried to find a Bohrbug your workflow has fallen victim to :D

At first glance, everything appears fine, like everything should work. The same goes for the second, third, and all the subsequent glances. However, thankfully, I decided to replace your `ls` with `ls -l`, and the issue suddenly became painfully obvious:

![FFS](https://github.com/Kir-Antipov/mc-publish/assets/40232067/c285e064-bb15-4fda-9d58-591e19d1bdc6)

These are not files. These are all directories :D

It seems that `dawidd6/action-download-artifact`, which you use, doesn't extract archives in place. Instead, it places their contents in a directory with the same name as the base name of the archive in question. So, for example, instead of:

```yaml
files: |
  AdvancedBackups-forge-1.7.10-${{ steps.version.outputs.VERSION }}.jar
```

You actually need:

```yaml
files: |
  **/AdvancedBackups-forge-1.7.10-${{ steps.version.outputs.VERSION }}.jar
```

Or, better yet, fix `dawidd6/action-download-artifact`'s output so that the files are actually located where we, as sane persons, expected them to be!

So, this is not really an issue with `mc-publish`, but I'm always glad to help :)

**Reply by HeatherComputer** [reply]
...wow...
I wonder if my previous curseforge and modrinth workflows just didn't particularly care then? This directory issue wouldn't be new, it should be rather old actually.

Thank you! 
*And of course I made some further mistakes that are gonna take a LOT of time to fix..*

**Reply by HeatherComputer** [reply]
![image](https://github.com/Kir-Antipov/mc-publish/assets/66441550/c3362434-142b-4d70-bc35-cbce95e6fac2)

And I've gotta wait for approval to fix any of them on curseforge, wonderful

**Reply by Kira-NT** [reply]
> I wonder if my previous curseforge and modrinth workflows just didn't particularly care then?

That's possible. It depends on the semantics of the tools you were using. I could start silently appending `**/` to all the people's globs to achieve the "find that thingy somewhere around here" effect, but I prefer explicitness in this case. As you can probably tell, it's better to upload nothing than to upload a wrong thing the wrong way *(yikes, best wishes with that. I once had to wait a week with CurseForge in a similar scenario...)*

**Reply by HeatherComputer** [reply]
Hehe, it took about an hour for them to get approved, then another half hour or so for me to fix the errors I made. Nice and quick in the grand scheme of things.

Your action worked wonderfully - though I did notice at one point it didn't log uploading one of the files to modrinth, despite the file actually being on modrinth when I checked. Harmless, but might be worth pointing out.

**Reply by Kira-NT** [reply]
> it took about an hour for them to get approved

Wow, in terms of CF, that's world record-level of fast xD

> though I did notice at one point it didn't log uploading one of the files to modrinth, despite the file actually being on modrinth when I checked

Huh? Could you, please, share a workflow run where this happened?

**Reply by HeatherComputer** [reply]
> Wow, in terms of CF, that's world record-level of fast xD

Certainly quicker than I expected, especially as I'm fairly certain Friday is their weekend..

> Huh? Could you, please, share a workflow run where this happened?

Apparently not - looking back, I can't even find it. Could well have been a github issue, I've seen the logs skip lines before...

Sorry haha
